activerecord/lib/active_record/asynchronous_queries_tracker.rb

```diff
@@ -50,7 +50,7 @@ def start_session
     end
 
     def finalize_session
-      @current_session&.finalize
+      @current_session.finalize if @current_session.respond_to?(:finalize)
```

Comment:
The point of using a null object is to avoid conditionals. In that case the `NullSession` should implement `finalize` and the `&` should be probably removed.

Reply:
That makes sense. I've added a noop `finalize` method to `NullSession`. I don't think it actually has to do anything for now?